lms/djangoapps/certificates/utils.py

```diff
@@ -250,3 +255,25 @@ def get_preferred_certificate_name(user):
         name_to_use = ''
 
     return name_to_use
+
+
+def get_certificate_configuration_from_credentials(course_id: CourseKey, mode: str) -> Optional[Response]:
+    """
+    Makes a request to the credentials service to get a certificate configuration of the course.
+    """
+    course_certificates_api_url = urljoin(f'{get_credentials_api_base_url()}/', 'course_certificates/')
+    api_url = urljoin(course_certificates_api_url, f'?course_id={course_id}&certificate_type={mode}')
+
+    try:
+        credentials_client = get_credentials_api_client(
+            User.objects.get(username=settings.CREDENTIALS_SERVICE_USERNAME)
+        )
+        response = credentials_client.get(api_url)
+    except Exception:   # pylint: disable=broad-except
+        log.warning(
+            'There is no certificate configuration in credentials for a course [%s] with a mod [%s]',
```

Comment:
You can't say that at all. After all, you do not understand whether there is a certificate or not. Maybe the network is down or the service is not responding. Everything is much more complicated here.

Reply:
And the second important question. What will you do when there are problems with the network? You then return NONE. Is this normal for your logic?

Reply:
Refactored.
If the response is None, the current configuration will remain from mongo.